src/lib/chain/offer/updateOffers.ts

```diff
@@ -0,0 +1,456 @@
+/**
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ComputeUnit } from "@fluencelabs/deal-ts-clients/dist/dealExplorerClient/types/schemes.js";
+import { color } from "@oclif/color";
+
+import { commandObj } from "../../commandObj.js";
+import { CLI_NAME } from "../../const.js";
+import { getDealClient, signBatch, populateTx } from "../../dealClient.js";
+import { uint8ArrayToHex } from "../../helpers/typesafeStringify.js";
+import { splitErrorsAndResults } from "../../helpers/utils.js";
+import { confirm } from "../../prompt.js";
+import {
+  cidStringToCIDV1Struct,
+  peerIdHexStringToBase58String,
+  cidHexStringToBase32,
+} from "../conversions.js";
+import { ptFormatWithSymbol } from "../currencies.js";
+import { assertProviderIsRegistered } from "../providerInfo.js";
+
+import {
+  type OffersArgs,
+  resolveOffersFromProviderConfig,
+  type EnsureOfferConfig,
+} from "./offer.js";
+
+export async function updateOffers(flags: OffersArgs) {
+  const offers = await resolveOffersFromProviderConfig(flags);
+  const offersFoundOnChain = filterOffersFoundOnChain(offers);
+  const populatedTxs = await populateUpdateOffersTxs(offersFoundOnChain);
+
+  const txsToSign = [
+    populatedTxs.flatMap(({ txsToRunBeforeOthers }) => {
+      return txsToRunBeforeOthers.map(({ tx }) => {
+        return tx;
+      });
+    }),
+    populatedTxs.flatMap(({ txs }) => {
+      return txs.map(({ tx }) => {
+        return tx;
+      });
+    }),
+  ].flat();
+
+  if (txsToSign.length === 0) {
+    commandObj.logToStderr("No changes found for selected offers");
+    return;
+  }
+
+  printOffersToUpdateInfo(populatedTxs);
+
+  if (
+    !(await confirm({
+      message: "Would you like to continue",
+      default: true,
+    }))
+  ) {
+    commandObj.logToStderr("Offers update canceled");
+    return;
```

Comment:
Will it return exit 1? I think it should 🤔

Reply:
it will not. I don't think it should. It's normal operation. You wanted to update. You inspected the diff. You decided not to proceed yourself